### PR TITLE
Fix shellcheck lints

### PR DESCRIPTION
**Pull request by ocefpaf**
In this PR:

- add shellcheck as pre-commit instead of GitHub Actions
- fix all lints

While submitting this PR I cannot stop thinking about this quote: https://github.com/koalaman/shellcheck#testimonials

### CODE REVIEW

**Comment by ocefpaf** [comment]
@kbailey-noaa this PR should fix the shell scripts failures you reported to me earlier today

@benjwadams can you review this one? Ideally we should remove the unused scripts and/or move them to [python if possible](https://github.com/koalaman/shellcheck#testimonials).

File: scripts/create_navoceano_symlinks.bash [hunk 1]
```diff
@@ -4,7 +4,7 @@
 # not have symlinks yet.
 shopt -s nullglob
 navo_dir=/data/data/priv_erddap/navoceano
-cd "$navo_dir"
+cd "$navo_dir" || exit
```
Comment (benjwadams) [inline scripts/create_navoceano_symlinks.bash:7]: Not a huge fan of this, but it's a temporary script if anything.

**Comment by benjwadams** [comment]
A lot of these scripts aren't used anymore.

**Comment by benjwadams** [comment]
Yes, I'll create a separate issue tomorrow.

**Pull request merged**